### PR TITLE
Ensure correct permissions for salt-cloud generated files

### PR DESCRIPTION
**Pull request by pprkut**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

salt-cloud creates the initial minion config and certificates with the permissions of the user it's connected as.
These checks make sure the permissions are the same as in a non salt-cloud setup.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by pprkut** [comment]
Not sure about the failing tests :-/

It looks like on those systems the pki-dir is different from /etc/salt/pki, but no config (that I can find) that would define it.
Am I missing something?

**Comment by noelmcloughlin** [comment]
@pprkut sorry for delay getting feedback.  Could you add `makedirs: true` to your file.directory states?

The CI error is : _No directory to create /etc/salt/pki/minion in_

**Comment by noelmcloughlin** [comment]
ping @pprkut ;-)

File: salt/minion.sls [hunk 1]
```diff
@@ -191,3 +191,51 @@ remove-macpackage-salt:
     - name: /tmp/salt.pkg
     - force: True
     {% endif %}
+
+permissions-minion-config:
+  file.managed:
+    - name: {{ salt_settings.config_path }}/minion
+    - user: root
+    - group: root
```
Comment (vutny) [inline salt/minion.sls:199]: On *BSD systems including MacOS, IIRC group `root` does not exist.

File: salt/minion.sls [hunk 2]
```diff
@@ -191,3 +191,51 @@ remove-macpackage-salt:
...
+    - mode: 640
```
Comment (vutny) [inline salt/minion.sls:200]: Also file mode is not supported on Windows.

**Review by vutny — CHANGES_REQUESTED** [review]
Hi @pprkut 
I think, those added file states should to be done with cross-platform support, now they have parameters supported on GNU/Linux only. Please find examples how to render states for different operating systems in the modified file.

**Comment by pprkut** [comment]
@vutny I think I have that covered now

**Comment by pprkut** [comment]
Hmm, the windows tests seem to stumble over the group now. I thought that was ignored on windows?

**Comment by noelmcloughlin** [comment]
Hello @pprkut 
Windows should be easy fix hopefully, see https://github.com/saltstack-formulas/docker-formula/pull/299/files

**Comment by noelmcloughlin** [comment]
I think Windows group needs  `{% if grains['kernel'] != 'Windows' %}` wrapped

**Comment by pprkut** [comment]
I see the windows states failing on this though:
```
The following packages failed to install/update: salt-minion-py3
```

**Comment by pprkut** [comment]
I'm afraid I'm a bit lost now on the windows failures :(
Since one of them succeeded, I'm assuming it *should* work fine generally. The remaining failures *look* unrelated to my changes, but I can't be sure. Can someone with more knowledge about windows have a look?

**Comment by myii** [comment]
> I'm afraid I'm a bit lost now on the windows failures :( Since one of them succeeded, I'm assuming it _should_ work fine generally. The remaining failures _look_ unrelated to my changes, but I can't be sure. Can someone with more knowledge about windows have a look?

@pprkut Yes, the failures are unrelated to your changes.  Our Windows Vagrant boxes were updated to `3004` recently, which have led to these failures.

@noelmcloughlin @vutny Once you're happy with the proposed changes, please feel free to merge.

**Comment by noelmcloughlin** [comment]
Thanks @vutny for review. Thanks @pprkut  for rescuing the PR.

**Pull request merged**